assimilation_code/programs/preprocess/preprocess.html

```diff
@@ -31,64 +31,79 @@ <H1>PROGRAM preprocess</H1>
-</ol>
+Preprocess is a DART-supplied preprocessor program.  Preprocess is used to
+insert observation specific code into DART at compile time.
+
+<p>
+In DART, forward operators are not specific to any one model. To achieve this
+separation between models and forward operators DART makes a distinction between
+an observation <em>type</em> and a physical <em>quantity</em>.
+For example, a radiosonde used to measure windspeed would be a <em>type</em> of
+observation. Zonal wind and meridional wind are <em>quantities</em> used
+to calculate windspeed. Specifying many observation types allows DART to be able
+to evaluate some observations and assimilate others even if the instruments
+measure the same quantity.
+
+<p>
+Preprocess takes user supplied observation and quantity files and combines them
+with template files to produce code for DART. Use the namelist option
+'obs_type_files' to specify the input observation files and  the namelist option
+'quantity_files' to specify the input quantity files.
+
+<p>
+<ul>
+<li>If no quantity files are given, a default list of quantities is used.
+<li>If no obs_type_files are given, only identity observations can be used in
+  the filter (i.e. the state variable values are directly observed;
+forward operator is an identity)
+</ul>
 
-<P>
-The DEFAULT files contain specially formatted comment lines which are
-used as markers to insert additional information into the templates
-and create the output files.
-<br>
-<br>
-If one or more special obs_def files are specified, information from these
-files is incorporated into the output obs_def and obs_kind modules and the
-observation types specified in the special obs_def files will be available
-for use by the filter programs.
-<br>
-<br>
-The special obs_def files must contain a number of sections of specially
-formatted F90 comments. They can also contain F90 code required for
-implementing operations needed for the observations. These sections are
-described in the documentation for the special obs_def modules, especially see
-<em class="file">obs_def_1d_state_mod.f90</em>.
-<br>
-<br> 
-The DEFAULT obs_kind and obs_def templates also require specially commented
-sections to indicate where code from the special obs_def modules is to be
-inserted. See the documentation for the DEFAULT_obs_kind_mod.F90 and the
-DEFAULT_obs_def_mod.F90 for details.
+<p>
+The template files <em class="file">DEFAULT_obs_def_mod.F90</em> and
+<em class="file">DEFAULT_obs_kind_mod.F90</em> contain specially formatted
+comment lines. These comment lines are used as markers to insert observation
+specific information. Prepreocess relies these comment lines being used <em>verbatim</em>.
+
+<p>
+There is no need to to alter <em class="file">DEFAULT_obs_def_mod.F90</em> or
+<em class="file">DEFAULT_obs_kind_mod.F90</em>. Detailed instructions for adding
+new observation types are in <a href="obs_def_mod.html">obs_def_mod</a>. New
```

Comment:
should the text for this link be: obs_def_mod.html? 
or it could be phrased:  are in the documentation file for <a href...